src/main/java/com/sivalabs/ft/features/domain/FavoriteFeatureService.java

```
@@ -0,0 +1,34 @@
+package com.sivalabs.ft.features.domain;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class FavoriteFeatureService {
+    private final FavoriteFeatureRepository favoriteFeatureRepository;
+    private final FeatureRepository featureRepository;
+
+    public FavoriteFeatureService(
+            FavoriteFeatureRepository favoriteFeatureRepository, FeatureRepository featureRepository) {
+        this.favoriteFeatureRepository = favoriteFeatureRepository;
+        this.featureRepository = featureRepository;
+    }
+
+    public void addFavoriteFeature(String userId, long featureId) {
+        // Check if the feature exists
+        featureRepository.findById(featureId).orElseThrow(() -> new EntityNotFoundException("Feature not found"));
+
+        // check if the favorite already exists
+        if (favoriteFeatureRepository.existsByUserIdAndFeatureId(userId, featureId)) {
+            return;
```

If userId + featureId combination already exists, then its a bad request. So throw Exception, instead of simply returning.

---

done